pkg/proxy/proxy.go

```diff
@@ -142,6 +151,181 @@ func (p *Proxy) allocatePort() (uint16, error) {
 
 var gcOnce sync.Once
 
+// LocalEndpointInfo fills the access log with the local endpoint info.
+func LocalEndpointInfo(info *accesslog.EndpointInfo, source ProxySource,
```

Comment:
All of these functions can be marked private

Reply:
LocalEndpointInfo needs redirect info for individual Kafka redirect as well as Oxyproxy,

Reply:
It's still weird to have `proxy.go` call an interface which requires `proxy.go` functionality itself again.

Looking at `LocalEndpointInfo()`, it requires `ProxySource` and the ID of the endpoint it is proxying for. The endpoint ID can be retrieved via the source using `source.GetID()`

Instead of writing into kafka and back into proxy, I'd rather add `getSource()` to `Redirect interface` which then just has to return the source